------------------------------------------------------------
Saving location: results/walker2d-medium-v2|exp_1|diffusion-ql|T-5|lr_decay|ms-offline|k-1|0
------------------------------------------------------------
/home/tori/miniconda3/envs/DQL/lib/python3.9/site-packages/gym/spaces/box.py:84: UserWarning: [33mWARN: Box bound precision lowered by casting to float32[0m
  logger.warn(f"Box bound precision lowered by casting to {self.dtype}")
24-12-17.14:40|Variant:
24-12-17.14:40|{
  "exp": "exp_1",
  "device": "cuda:0",
  "env_name": "walker2d-medium-v2",
  "dir": "results",
  "seed": 0,
  "num_steps_per_epoch": 1000,
  "batch_size": 256,
  "lr_decay": true,
  "early_stop": true,
  "save_best_model": false,
  "discount": 0.99,
  "tau": 0.005,
  "T": 5,
  "beta_schedule": "vp",
  "algo": "ql",
  "ms": "offline",
  "dual_diffusion": false,
  "output_dir": "results",
  "num_epochs": 100,
  "eval_freq": 50,
  "eval_episodes": 10,
  "lr": 0.0003,
  "eta": 1.0,
  "max_q_backup": false,
  "reward_tune": "no",
  "gn": 1.0,
  "top_k": 1,
  "version": "Diffusion-Policies-RL",
  "state_dim": 17,
  "action_dim": 6,
  "max_action": 1.0
}
------------------------------------------------------------
Env: walker2d-medium-v2, state_dim: 17, action_dim: 6, dual_diffusion: False
------------------------------------------------------------
load datafile: 100%|â–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆ| 21/21 [00:01<00:00, 16.54it/s]
------------------------------------------------------------
Loaded buffer
------------------------------------------------------------
******************************************************************************************
Training Start
******************************************************************************************
******************************************************************************************
Train step: 50000
******************************************************************************************
/home/tori/miniconda3/envs/DQL/lib/python3.9/site-packages/numpy/core/fromnumeric.py:3504: RuntimeWarning: Mean of empty slice.
  return _methods._mean(a, axis=axis, dtype=dtype,
/home/tori/miniconda3/envs/DQL/lib/python3.9/site-packages/numpy/core/_methods.py:129: RuntimeWarning: invalid value encountered in scalar divide
  ret = ret.dtype.type(ret / rcount)
Traceback (most recent call last):
  File "/home/tori/Documents/research/Diffusion-Policies-for-Offline-RL/main.py", line 280, in <module>
    train_agent(env,
  File "/home/tori/Documents/research/Diffusion-Policies-for-Offline-RL/main.py", line 109, in train_agent
    'Critic Loss': np.mean(loss_metric['critic_loss'])}, step=(curr_epoch*args.num_step_per_epoch))
AttributeError: 'Namespace' object has no attribute 'num_step_per_epoch'
